austraits_build_tutorial.qmd

```diff
@@ -0,0 +1,625 @@
+austraits::lookup_trait(austraits, "photo")
+```
+
+### Filtering data
+
+There are a trio of austraits.R functions to filter data by `trait_name`, `taxon_name` or `dataset_id`:
+
+```{r, eval = FALSE}
+leaf_mass_per_area_values <-
+  austraits %>% 
+     austraits::extract_trait(trait_names = c("leaf_mass_per_area"))
+
+Westoby_2014_values <-
+  austraits %>%
+     austraits::extract_dataset("Westoby_2014")
+
+Eucalyptus_values <-
+  austraits %>%
+     austraits::extract_taxa(genus = "Eucalyptus")
+```
+
+You can alternatively simply filter by whatever combination of values you are searching for:
+
+```{r, eval = FALSE}
+my_data <-
+  austraits$traits %>%
+    dplyr::filter(stringr::str_detect(taxon_name, "Eucalyptus")) %>%
+    dplyr::filter(trait_name %in% c("leaf_mass_per_area", "leaf_length", "leaf_thickness")) %>%
+    dplyr::filter(basis_of_record == "field")
+```
+
+## Wrangling data: joining tables
+
+### Locations
+
+To just add location name, latitude, and longitude:
+
+```{r, eval = FALSE}
+traits_with_lat_long <- 
+  (austraits %>% austraits::join_locations())$traits
+```
+
+To add all location properties:
+
+```{r, eval = FALSE}
+locations <- 
+  austraits$locations %>%
+  tidyr::pivot_wider(names_from = location_property, values_from = value)
+
+traits_with_locations <- 
+  austraits$traits %>% 
+  dplyr::left_join(locations, by = c("dataset_id", "location_id"))
+```
+
+The number of location properties is of course long. You'll possible want to check which location properties you care about and just merge those:
+
+```{r, eval = FALSE}
+locations_properties <- 
+  austraits$locations %>% 
+  dplyr::distinct(location_property)
+  
+locations <- 
+  austraits$locations %>%
+  dplyr::filter(location_property %in% c("latitude (deg)", "longitude (deg)", "description", "aridity index (MAP/PET)", "soil type", "fire history")) %>%
+  tidyr::pivot_wider(names_from = location_property, values_from = value)
+
+traits_with_locations <- 
+  austraits$traits %>% 
+  dplyr::left_join(locations, by = c("dataset_id", "location_id"))
+```
+
+### Contexts
+
+Joining the context table to the traits table is best done using the austraits.R function, as there are 5 identifiers in the traits table (`entity_context_id`, `method_context_id`, `temporal_context_id`, `plot_context_id` and `treatment_context_id`) that link to two columns in the contexts table (`link_vals` and `link_id`).
+
+```{r, eval = FALSE}
+traits_with_contexts <- 
+  (austraits %>% austraits::join_contexts())$traits %>% 
+  dplyr::select(-dplyr::all_of(c("temporal_context_id", "method_context_id", "treatment_context_id", "entity_context_id", "plot_context_id")))
+```
+
+If you wish to include fewer context properties, you have to manually select the columns you wish to retain/lose.
+
+### Methods
+
+To just add trait methods:
+
+```{r, eval = FALSE}
+traits_with_methods <- 
+  (austraits %>% austraits::join_methods())$traits
```

Comment:
I would seperate this into two steps

```
traits_with_methods <- austraits %>% austraits::join_methods() 

traits_with_methods$traits
```

Similar for taxa below

Reply:
I agree! I was following examples in austraits vignettes and thought this was what was "preferred" - I find 2 steps much easier